spec/requests/donation_sites_requests_spec.rb

```diff
@@ -32,5 +32,30 @@
         it { is_expected.to be_successful }
       end
     end
+    describe 'GET #index' do
+      let!(:active_donation_site) { create(:donation_site, organization: @organization, name: "An Active Site") }
+      let!(:inactive_donation_site) { create(:donation_site, organization: @organization, active: false, name: "An Inactive Site") }
+
+      it "should show all/only active donation sites with deactivate buttons" do
+        get donation_sites_path(default_params)
+        page = Nokogiri::HTML(response.body)
+        expect(response.body).to include("An Active Site")
+        expect(response.body).not_to include("An Inactive Site")
+        button1 = page.css(".btn[href='/#{@organization.short_name}/donation_sites/#{active_donation_site.id}/deactivate']")
+        expect(button1.text.strip).to eq("Deactivate")
+        expect(button1.attr('class')).not_to match(/disabled/)
+      end
+    end
+
+    describe 'DELETE #deactivate' do
+      it 'should be able to deactivate an item' do
+        donation_site = create(:donation_site, organization: @organization, active: true, name: "to be deactivated")
+        params = default_params.merge(id: donation_site.id)
+
+        expect { delete deactivate_donation_site_path(params) }.to change { donation_site.reload.active }.from(true).to(false)
+          .and change { DonationSite.active.count }.by(-1)
```

Comment:
This can be insidious - it could change the active count by 1 because it deactivated it, or because it deleted it :grin: The right thing to do is just to check the reloaded donation\_site to see if it is now inactive.

Reply:
Or I could add a check that it doesn't change DonationSite.count.